src/imageLoader/wadors/getPixelData.js

```diff
@@ -34,49 +34,53 @@ function getPixelData (uri, imageId, mediaType = 'application/octet-stream') {
     accept: mediaType
   };
 
-  return new Promise((resolve, reject) => {
-    const loadPromise = xhrRequest(uri, imageId, headers);
+  const loadObject = xhrRequest(uri, imageId, headers);
 
-    loadPromise.then(function (imageFrameAsArrayBuffer/* , xhr*/) {
+  return {
+    promise: new Promise((resolve, reject) => {
 
-      // request succeeded, Parse the multi-part mime response
-      const response = new Uint8Array(imageFrameAsArrayBuffer);
+      loadObject.promise.then(function (imageFrameAsArrayBuffer/* , xhr*/) {
 
-      // First look for the multipart mime header
-      const tokenIndex = findIndexOfString(response, '\r\n\r\n');
+        // request succeeded, Parse the multi-part mime response
+        const response = new Uint8Array(imageFrameAsArrayBuffer);
 
-      if (tokenIndex === -1) {
-        reject(new Error('invalid response - no multipart mime header'));
-      }
-      const header = uint8ArrayToString(response, 0, tokenIndex);
-      // Now find the boundary  marker
-      const split = header.split('\r\n');
-      const boundary = findBoundary(split);
+        // First look for the multipart mime header
+        const tokenIndex = findIndexOfString(response, '\r\n\r\n');
 
-      if (!boundary) {
-        reject(new Error('invalid response - no boundary marker'));
-      }
-      const offset = tokenIndex + 4; // skip over the \r\n\r\n
-
-      // find the terminal boundary marker
-      const endIndex = findIndexOfString(response, boundary, offset);
+        if (tokenIndex === -1) {
+          reject(new Error('invalid response - no multipart mime header'));
+        }
+        const header = uint8ArrayToString(response, 0, tokenIndex);
+        // Now find the boundary  marker
+        const split = header.split('\r\n');
+        const boundary = findBoundary(split);
 
-      if (endIndex === -1) {
-        reject(new Error('invalid response - terminating boundary not found'));
-      }
+        if (!boundary) {
+          reject(new Error('invalid response - no boundary marker'));
+        }
+        const offset = tokenIndex + 4; // skip over the \r\n\r\n
 
-      // Remove \r\n from the length
-      const length = endIndex - offset - 2;
+        // find the terminal boundary marker
+        const endIndex = findIndexOfString(response, boundary, offset);
```

Comment:
Why did this stuff change? Could you rebase your commits on top of master? It doesn't look like this is meant to be part of your changes? If you intended to introduce these changes, could you split them into another PR, since it's unrelated functionality?

Reply:
I did the rebase, but this github diff is still showing differences with master that don't exist.